packages/nimble-components/src/table/tests/table.spec.ts

```diff
@@ -575,4 +580,361 @@ describe('Table', () => {
             );
         });
     });
+
+    describe('action menu', () => {
+        function createAndSlotMenu(slot: string): {
+            menu: Menu,
+            items: MenuItem[]
+        } {
+            const menu = document.createElement('nimble-menu');
+            menu.slot = slot;
+
+            const menuItem1 = document.createElement('nimble-menu-item');
+            menuItem1.textContent = 'menu item 1';
+            menu.appendChild(menuItem1);
+
+            const menuItem2 = document.createElement('nimble-menu-item');
+            menuItem2.textContent = 'menu item 2';
+            menu.appendChild(menuItem2);
+
+            const menuItem3 = document.createElement('nimble-menu-item');
+            menuItem3.textContent = 'menu item 3';
+            menu.appendChild(menuItem3);
+
+            element.appendChild(menu);
+
+            return { menu, items: [menuItem1, menuItem2, menuItem3] };
+        }
+
+        function createTableListener(eventName: string): {
+            promise: Promise<void>,
+            spy: jasmine.Spy
+        } {
+            const spy = jasmine.createSpy();
+            return {
+                promise: new Promise(resolve => {
+                    const handler = (...args: unknown[]): void => {
+                        element.removeEventListener(eventName, handler);
```

Comment:
Neat utility! If you change element to be a parameter could use generically. Could add to https://github.com/ni/nimble/blob/main/packages/nimble-components/src/utilities/tests/component.ts

Reply:
I'm glad you like it -- I'm pretty sure you were the one that suggested it when I needed it for the menu button tests last year 😃 

That does bring up a good point, though. I copied it from the menu button tests, and I think the pattern was also used in the banner tests. It's about time we just make it a common utility.